spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkScanBuilder.java

```diff
@@ -153,8 +158,90 @@ private Schema schemaWithMetadataColumns() {
 
   @Override
   public Scan build() {
```

Comment:
@rdblue, I am not entirely happy with this place but it is probably better than using a context.